File: egov-lowdata-suite/docs/portfolio/capture_screenshots.js
const { chromium } = require('playwright');
const fs = require('fs');
const path = require('path');

const BASE = 'http://localhost:8080';
const OUT = path.join(__dirname, 'screenshots');

const shots = [
  { file: '01-login.png', url: `${BASE}/auth/login.do`, wait: 1000 },
  { file: '02-dashboard.png', url: `${BASE}/home/dashboard.do`, login: true, wait: 2000 },
  { file: '03-project-list.png', url: `${BASE}/projects/list.do`, login: true, wait: 1500 },
  { file: '04-project-detail.png', url: `${BASE}/projects/d1000000-0000-4000-8000-000000000001/detail.do`, login: true, wait: 1500 },
  { file: '05-project-analytics.png', url: `${BASE}/projects/d1000000-0000-4000-8000-000000000001/analytics.do`, login: true, wait: 2500 },
  { file: '06-deleted-projects.png', url: `${BASE}/projects/deleted/list.do`, login: true, wait: 1500 },
];

async function login(page) {
  await page.goto(`${BASE}/auth/login.do`, { waitUntil: 'networkidle' });
  await page.fill('#username', 'admin');
  await page.fill('#password', 'admin');
  await page.click('button[type="submit"], input[type="submit"]');
  await page.waitForLoadState('networkidle');
}

(async () => {
  fs.mkdirSync(OUT, { recursive: true });
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({ viewport: { width: 1440, height: 900 } });
  const page = await context.newPage();

  let loggedIn = false;
  for (const shot of shots) {
    if (shot.login && !loggedIn) {
      await login(page);
      loggedIn = true;
    }
    await page.goto(shot.url, { waitUntil: 'networkidle' });
    await page.waitForTimeout(shot.wait || 1000);
    await page.screenshot({ path: path.join(OUT, shot.file), fullPage: true });
    console.log('saved', shot.file);
  }

  await browser.close();
})();

File: egov-lowdata-suite/docs/portfolio/capture_screenshots.py
#!/usr/bin/env python3
"""Capture portfolio screenshots from running LINC gov-portal."""

from pathlib import Path
from playwright.sync_api import sync_playwright

BASE = "http://localhost:8080"
OUT = Path(__file__).resolve().parent / "screenshots"
OUT.mkdir(parents=True, exist_ok=True)

SHOTS = [
    ("01-login.png", f"{BASE}/login.do", False, 1200),
    ("02-dashboard.png", f"{BASE}/dashboard.do", True, 2500),
    ("03-project-list.png", f"{BASE}/projects/list.do", True, 1800),
    (
        "04-project-detail.png",
        f"{BASE}/projects/d1000000-0000-4000-8000-000000000001/detail.do",
        True,
        1800,
    ),
    (
        "05-project-analytics.png",
        f"{BASE}/projects/d1000000-0000-4000-8000-000000000001/analytics.do",
        True,
        3500,
    ),
    ("06-deleted-projects.png", f"{BASE}/projects/deleted/list.do", True, 1800),
]


def login(page):
    page.goto(f"{BASE}/login.do", wait_until="networkidle")
    page.fill("#username", "admin")
    page.fill("#password", "admin")
    submit = page.locator('button[type="submit"], input[type="submit"]').first
    submit.click()
    page.wait_for_load_state("networkidle")


def main():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(viewport={"width": 1440, "height": 900})
        page = context.new_page()
        logged_in = False

        for filename, url, needs_login, wait_ms in SHOTS:
            if needs_login and not logged_in:
                login(page)
                logged_in = True
            page.goto(url, wait_until="networkidle")
            page.wait_for_timeout(wait_ms)
            target = OUT / filename
            page.screenshot(path=str(target), full_page=True)
            print(f"saved {target.name}")

        browser.close()


if __name__ == "__main__":
    main()
